tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_BjorhusImpl.cpp

```diff
@@ -0,0 +1,848 @@
+        inverse_spacetime_metric,
+    const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>&
+        three_index_constraint,
+    const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_psi,
+    const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_minus,
+    const tnsr::a<DataVector, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_plus,
+    const tnsr::a<DataVector, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_minus,
+    const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>& phi,
+    const tnsr::ijaa<DataVector, VolumeDim, Frame::Inertial>& d_phi,
+    const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>& d_pi,
+    const tnsr::a<DataVector, 3, Frame::Inertial>& char_speeds) noexcept {
+  std::array<DataVector, 4> char_speed_array{
+      get<0>(char_speeds), get<1>(char_speeds), get<2>(char_speeds),
+      get<3>(char_speeds)};
+  auto dt_v_minus =
+      make_with_value<tnsr::aa<DataVector, VolumeDim, Frame::Inertial>>(
+          get(gamma2), 0.);
+  GeneralizedHarmonic::BoundaryConditions::Bjorhus::
+      add_constraint_preserving_physical_terms_to_dt_v_minus<VolumeDim,
+                                                             DataVector>(
+          make_not_null(&dt_v_minus), gamma2, inertial_coords,
+          unit_interface_normal_one_form, unit_interface_normal_vector,
+          spacetime_unit_normal_vector, incoming_null_one_form,
+          outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+          projection_ab, projection_Ab, projection_AB, inverse_spatial_metric,
+          extrinsic_curvature, spacetime_metric, inverse_spacetime_metric,
+          three_index_constraint, char_projected_rhs_dt_v_psi,
+          char_projected_rhs_dt_v_minus, constraint_char_zero_plus,
+          constraint_char_zero_minus, phi, d_phi, d_pi, char_speed_array);
+  return dt_v_minus;
+}
+
+template <size_t VolumeDim>
+void test_add_constraint_preserving_terms_to_dt_v_minus(
+    const size_t grid_size_each_dimension) noexcept {
+  pypp::check_with_random_values<1>(
+      &wrapper_func_cp_v_minus<VolumeDim>,
+      "Evolution.Systems.GeneralizedHarmonic.BoundaryConditions.Bjorhus",
+      "add_constraint_preserving_terms_to_dt_v_minus", {{{-1., 1.}}},
+      DataVector(grid_size_each_dimension));
+}
+
+template <size_t VolumeDim>
+void test_add_constraint_preserving_physical_terms_to_dt_v_minus(
+    const size_t grid_size_each_dimension) noexcept {
+  pypp::check_with_random_values<1>(
+      &wrapper_func_cpp_v_minus<VolumeDim>,
+      "Evolution.Systems.GeneralizedHarmonic.BoundaryConditions.Bjorhus",
+      "add_constraint_preserving_physical_terms_to_dt_v_minus", {{{-1., 1.}}},
+      DataVector(grid_size_each_dimension));
+}
+}  // namespace
+
+namespace {
+using frame = Frame::Inertial;
+constexpr size_t VolumeDim = 3;
+
+void test_constraint_preserving_physical_bjorhus_v_minus_vs_spec_3d(
```

Comment:
I don't see a need for this function if we have a pypp test

Reply:
the pypp test tests for an agreement between two implementations of the same terms. This test tests for an agreement between the implementation here and that in SpEC where the same has been tested in production simulations. I supposes this test can be removed post-facto once we have performed similarly confident evolutions with spectre, but I'd advocate to have this test included for now.